WhatsApp/app/src/main/java/com/example/whatsapp/ui/fragments/home/calls/CallsFragmentVM.kt

```diff
@@ -0,0 +1,83 @@
+package com.example.whatsapp.ui.fragments.home.calls
+
+import com.example.firestorerepository.datatypes.CallHistory
+import com.example.firestorerepository.repositories.UserRepositoryImpl
+import com.example.firestorerepository.repositories.chat.ChatRepositoryImpl
+import com.example.whatsapp.base.BaseViewModel
+import dagger.hilt.android.lifecycle.HiltViewModel
+import io.reactivex.rxjava3.core.Observable
+import io.reactivex.rxjava3.subjects.PublishSubject
+import javax.inject.Inject
+
+@HiltViewModel
+class CallsFragmentVM @Inject constructor(
+    private val userRepo: UserRepositoryImpl,
```

Comment:
Heyy, we don't want to inject the Implementation of the Repo. We'd inject its interface instead :)